lib-model/lib-model-consume-cli/src/main/java/com/uber/nullaway/libmodel/LibModelInfoExtractor.java

```diff
@@ -0,0 +1,287 @@
+
+    projectRoot
+        .getSourceRoots()
+        .forEach(
+            sourceRoot -> {
+              try {
+                sourceRoot.parse("", mc);
+              } catch (IOException e) {
+                System.err.println("IOException: " + e);
+              }
+            });
+  }
+
+  public void writeToAstubx(String outputPath, Map<String, MethodAnnotationsRecord> methodRecords) {
+    Map<String, String> importedAnnotations =
+        ImmutableMap.<String, String>builder()
+            .put("Nonnull", "javax.annotation.Nonnull")
+            .put("Nullable", "javax.annotation.Nullable")
+            .build();
+    DataOutputStream dos;
+    try {
+      Path opPath = Paths.get(outputPath);
+      Files.createDirectories(opPath.getParent());
+      dos = new DataOutputStream(Files.newOutputStream(opPath));
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+    if (!methodRecords.isEmpty()) {
+      try {
+        StubxWriter.write(
+            dos, importedAnnotations, new HashMap<>(), new HashMap<>(), methodRecords);
+        dos.close();
+      } catch (IOException e) {
+        throw new RuntimeException(e);
+      }
+    }
+  }
+
+  public Path dirnameToPath(String dir) {
+    File f = new File(dir);
+    if (!f.exists()) {
+      System.err.printf("Directory %s (%s) does not exist.%n", dir, f);
+      System.exit(1);
+    }
+    if (!f.isDirectory()) {
+      System.err.printf("Not a directory: %s (%s).%n", dir, f);
+      System.exit(1);
+    }
+    String absoluteDir = f.getAbsolutePath();
+    if (absoluteDir.endsWith("/.")) {
+      absoluteDir = absoluteDir.substring(0, absoluteDir.length() - 2);
+    }
+    return Paths.get(absoluteDir);
+  }
+
+  private static class AnnotationCollectorCallback implements SourceRoot.Callback {
+
+    private final AnnotationCollectionVisitor mv;
+    private Map<String, MethodAnnotationsRecord> methodRecords;
+
+    public AnnotationCollectorCallback(Map<String, MethodAnnotationsRecord> methodRecords) {
+      this.methodRecords = methodRecords;
+      this.mv = new AnnotationCollectionVisitor();
+    }
+
+    @Override
+    public Result process(Path localPath, Path absolutePath, ParseResult<CompilationUnit> result) {
+      Result res = Result.SAVE;
+      Optional<CompilationUnit> opt = result.getResult();
+      if (opt.isPresent()) {
+        CompilationUnit cu = opt.get();
+        visitAll(cu);
+      }
+      return res;
+    }
+
+    public void visitAll(Node rootNode) {
+      if (rootNode == null) {
+        return;
+      }
+      ArrayDeque<Node> stack = new ArrayDeque<>();
+      stack.addFirst(rootNode);
+      while (!stack.isEmpty()) {
+        Node current = stack.removeFirst();
+        current.accept(mv, this.methodRecords);
+        List<Node> children = current.getChildNodes();
+        if (children != null) {
+          for (Node child : children) {
+            stack.addFirst(child);
+          }
+        }
+      }
+    }
+  }
```

Comment:
I don't think we need this logic to recurse?  The default `VoidVisitorAdapter` behavior is to visit children.  So if you just call `super.visit()` at the end of each of your `visit()` methods, then everything should get visited by default